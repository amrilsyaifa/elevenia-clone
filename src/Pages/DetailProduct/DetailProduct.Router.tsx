import React from 'react';
import { Route } from 'wouter';
import PageRouter, { IPageRouter } from '../../Routers/PageRouter';
import DetailProduct from './DetailProduct';
import Checkout from './Pages/Checkout/Checkout';

const DetailProductRouter: React.FC<IPageRouter> = (props) => {
  return (
    <PageRouter {...props}>
      <Route path="/detail/:id">
        <DetailProduct />
      </Route>
      <Route path="/checkout">
        <Checkout />
      </Route>
    </PageRouter>
  );
};

export default DetailProductRouter;
