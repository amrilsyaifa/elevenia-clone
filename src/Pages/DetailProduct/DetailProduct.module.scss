@import 'src/Reusables/Styles/Variables';

.title {
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 20px;
}
.container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  .wrap-image {
    display: inherit;
    flex-direction: column;
    .wrap-box {
      display: inherit;
      flex-direction: row;
      margin-top: 10px;
      .card {
        height: 80px;
        width: 150px;
        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
        text-align: center;
        font-size: 12px;
        .row {
          display: flex;
          flex-direction: row;
          align-items: center;
          .spacer-left {
            padding-left: 5px;
          }
        }
        .total {
          width: 100%;
        }
      }
    }
  }
  .wrap-description {
    display: inherit;
    flex-direction: column;
    width: 65%;
    .row {
      display: inherit;
      flex-direction: row;
      justify-items: center;
      .title {
        font-size: 14px;
        width: 200px;
        font-weight: normal;
      }
      .price {
        color: $red;
        font-size: 18px;
        font-weight: bold;
      }
    }
    .description {
      display: inherit;
      flex-direction: column;
      margin-bottom: 20px;
      div:nth-child(1) {
        font-weight: bold;
        font-size: 16px;
      }
      div:nth-child(2) {
        font-weight: normal;
        font-size: 14px;
        color: $grey;
      }
    }
  }
}
