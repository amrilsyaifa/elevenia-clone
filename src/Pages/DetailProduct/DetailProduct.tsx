import React from 'react';
import { Button } from 'antd';
import { useRoute } from 'wouter';
import { useAppSelector, useAppDispatch } from 'src/Hooks/useReduxHooks';
import { addDataCheckout, removeDataCheckout } from 'src/Store/Reducers/Checkout.Reducers';

const DetailProduct = () => {
  const data = useAppSelector((state) => state.checkout.data);
  const dispatch = useAppDispatch();

  const tempData = {
    url: 'https://ant.design/components/button/',
    prdNm: 'prdNm',
    sellerPrdCd: 'sellerPrdCd',
    selPrc: 'selPrc',
    stock: 'stock',
    prdSelQty: 'prdSelQty'
  };
  return (
    <div>
      <div>DetailProduct</div>
      <Button type="primary" onClick={() => dispatch(addDataCheckout(tempData))}>
        Add to Chart
      </Button>
      <Button type="primary" danger onClick={() => dispatch(removeDataCheckout(data.length - 1))}>
        Remove to Chart
      </Button>
    </div>
  );
};

export default DetailProduct;
