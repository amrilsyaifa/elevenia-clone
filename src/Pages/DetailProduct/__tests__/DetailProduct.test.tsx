import { render, cleanup } from 'src/Reusables/Mocks/test-utils';
import React from 'react';
import { MatchMediaObject } from 'src/Reusables/Mocks/MatchMediaObjectTest';
import DetailProduct from '../DetailProduct';

afterEach(cleanup);
beforeEach(() => {
  MatchMediaObject();
});

describe('DetailProduct', () => {
  test('DetailProduct Render Snapshoot', () => {
    const { container } = render(<DetailProduct />);
    expect(container).toMatchSnapshot();
  });
});
