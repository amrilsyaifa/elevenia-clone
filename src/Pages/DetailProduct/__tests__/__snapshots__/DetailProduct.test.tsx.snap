// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailProduct DetailProduct Render Snapshoot 1`] = `
<div>
  <div>
    <div>
      DetailProduct
    </div>
    <button
      class="ant-btn ant-btn-primary"
      type="button"
    >
      <span>
        Add to Chart
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-dangerous"
      type="button"
    >
      <span>
        Remove to Chart
      </span>
    </button>
  </div>
</div>
`;
